src/components/modals/ReconfigureFCModal.tsx

```diff
@@ -184,70 +191,44 @@ export default function ReconfigureFCModal({
       })
     }
   }, [
+    currentFC,
+    queuedFC,
+    currentPayoutMods,
+    queuedPayoutMods,
+    currentTicketMods,
+    queuedTicketMods,
     dispatch,
-    fundingCycle,
-    payoutMods,
-    ticketMods,
     ticketingForm,
     restrictedActionsForm,
     isPreviewMode,
     visible,
   ])
 
   async function reconfigure() {
-    if (
-      !transactor ||
-      !contracts?.TerminalV1_1 ||
-      !projectId ||
-      !terminal?.version
-    )
-      return
-
     setLoading(true)
 
-    const properties: Record<keyof FCProperties, string> = {
-      target: editingFC.target.toHexString(),
-      currency: editingFC.currency.toHexString(),
-      duration: editingFC.duration.toHexString(),
-      discountRate: editingFC.discountRate.toHexString(),
-      cycleLimit: BigNumber.from(0).toHexString(),
-      ballot: editingFC.ballot,
-    }
-
-    const metadata: Omit<FundingCycleMetadata, 'version'> = {
-      reservedRate: editingFC.reserved.toNumber(),
-      bondingCurveRate: editingFC.bondingCurveRate.toNumber(),
-      reconfigurationBondingCurveRate: editingFC.bondingCurveRate.toNumber(),
-      payIsPaused: editingFC.payIsPaused,
-      ticketPrintingIsAllowed: editingFC.ticketPrintingIsAllowed,
-      treasuryExtension: constants.AddressZero,
-    }
-
-    transactor(
-      terminal.version === '1.1'
-        ? contracts.TerminalV1_1
-        : contracts.TerminalV1,
-      'configure',
-      [
-        projectId.toHexString(),
-        properties,
-        metadata,
-        editingPayoutMods.map(m => ({
-          preferUnstaked: false,
-          percent: BigNumber.from(m.percent).toHexString(),
-          lockedUntil: BigNumber.from(m.lockedUntil ?? 0).toHexString(),
-          beneficiary: m.beneficiary || constants.AddressZero,
-          projectId: m.projectId || BigNumber.from(0).toHexString(),
-          allocator: constants.AddressZero,
-        })),
-        editingTicketMods.map(m => ({
-          preferUnstaked: false,
-          percent: BigNumber.from(m.percent).toHexString(),
-          lockedUntil: BigNumber.from(m.lockedUntil ?? 0).toHexString(),
-          beneficiary: m.beneficiary || constants.AddressZero,
-          allocator: constants.AddressZero,
-        })),
-      ],
+    configureProjectTx(
+      {
+        fcProperties: {
+          target: editingFC.target,
+          currency: editingFC.currency,
+          duration: editingFC.duration,
+          discountRate: editingFC.discountRate,
+          cycleLimit: BigNumber.from(0), // TODO support in UI
```

Comment:
**question**: what does `cycleLimit` do?

Reply:
it's one of the funding cycle properties that the UI has never made use of https://docs.juicebox.money/protocol-v1/ticketbooth/funding-cycles#write

Reply:
Oh cool I see, so like a time-bound config that will auto-revert. I like.